app/controllers/api/base_provider_controller.rb

```diff
@@ -0,0 +1,14 @@
+module Api
+  class BaseProviderController < BaseController
+    def options
+      if (id = params["id"])
+        action = (@req.option_action || :update).to_sym
+        render_resource_options(id, action)
+      elsif (ems_id = params["ems_id"])
+        render_create_resource_options(ems_id)
+      else
+        super
+      end
+    end
```

Comment:
alternatively, we could make this change in the base controller.

I had higher ambitions for the `BaseProviderController` but this seemed good for now.